### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by salamf**
General overview of homepage with sample data for placeholders.

Screenshot 2024-02-11 at 2 47 31 PM

### CODE REVIEW

File: app/app/(tabs)/_layout.tsx [hunk 1]
```diff
@@ -25,7 +26,7 @@ export default () => {
       }}
     >
       <Tabs.Screen
-        name="App"
+        name="HomePage"
```
Comment (abraham-rostampoor) [inline app/app/(tabs)/_layout.tsx:29]: Great work, @salamf! I tested your work using the Xcode simulator on iOS devices (iPhone 15) and Android Studio on Android devices. However, there is an inconsistency in the navigation bar layout - there is a missing icon on the Android device, whereas it is visible on iOS. Additionally, I think we need to add a few images of food items and the names of restaurants to the homepage.
Screenshot 2024-02-13 at 12 37 13 PM

File: app/components/FoodSection.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { View, Text, StyleSheet, ScrollView } from "react-native";
+import { colors, getStyles } from "../constants/styles/global";
+
+// Will Change based off data model
+interface FoodItem {
```
Comment (abraham-rostampoor) [inline app/components/FoodSection.tsx:6]: Checkout @johnmck95's pull request for the type.ts file (https://github.com/SENG480a-NutriDine/mobile/pull/31/files#diff-a78b65a9a74e4f0240a068d4c67cd1a436e1dbc0d923dbd307f720e76b987216R1). We will be using the data format defined in this file. For instance, the type.ts file does not have a price attribute for FoodItem.

File: app/app/(tabs)/HomePage.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import { colors, getStyles } from "../../constants/styles/global";
+
+
+const HomePage = () => {
+
+    const { theme } = getStyles();
+
+    const sampleRestaurants = [
+        {
+            "name": "The Gourmet Hut",
```
Comment (abraham-rostampoor) [inline app/app/(tabs)/HomePage.tsx:14]: To populate the Home page, you can utilize the Firebase database to fetch the required data.
Screenshot 2024-02-13 at 12 58 48 PM

Comment (johnmck95) [reply app/app/(tabs)/HomePage.tsx:14]: My PR was just merged - on latest main you'll see a `useRestaurants` hook you can use to handle the fetch.

For the time being, restaurants & menus need to be manually added via the firebase console, only foods can be submitted through the form

File: app/components/CategorySection.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    /* this will later be extracted from db maybe? */
+    const categories = [
+        "High-Protein",
+        "Low-Calorie",
+        "Healthy",
+        "Vegan",
+        "Vegitarian",
+        "Halal",
+    ];
```
Comment (johnmck95) [inline app/components/CategorySection.tsx:16]: You'll be able to find the following under `food.nutrionalData`:

```
  isGlutenFree: boolean;
  isDairyFree: boolean;
  isVegetarian: boolean;
  isVegan: boolean;
  isEstimate: boolean;
  hasFreshFruit: boolean;
  hasFreshVegetables: boolean;
  ```

File: app/components/FoodSection.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        marginRight: 20,
+        padding: 15,
+        justifyContent: 'space-between',
+    },
+    itemHeader: {
+        fontSize: 20,
+    },
+    itemDescription: {
+        fontSize: 15,
+        color: "#666",
```
Comment (johnmck95) [inline app/components/FoodSection.tsx:76]: To keep the light/dark mode function usable, we should be assigning all colors like so:

` color: {colors[theme].button.text}`

Feel free to add to global.ts!

**Review by johnmck95 — COMMENTED** [review]
Overall looks great! Couple high-level questions for ya
1) What is the search bar going to do once it gets hooked up? Wondering how it will work from a user and a developer POV. (child components filtering parent data is never an easy implementation). Also wondering if this 'home page' will be for scrolling by menus/restaurants and finding ideas, while another page will offer dedicated macronutrient filters or something??
2) What is the down-caret in the layout footer doing? Where does it take you?

File: app/app/(tabs)/_layout.tsx (hunk 1 above)
Comment (salamf) [reply app/app/(tabs)/_layout.tsx:29]: I think that last icon is a placeholder indicating that there is no icon for that page (that icon leads to a test page that we will remove) - that's why its different on iOS and android.

File: app/components/FoodSection.tsx (hunk 2 above)
Comment (salamf) [reply app/components/FoodSection.tsx:6]: Yes I will update that, that was just temporary until we got the firebase stuff set up

File: app/app/(tabs)/HomePage.tsx (hunk 3 above)
Comment (salamf) [reply app/app/(tabs)/HomePage.tsx:14]: Sounds good John! I'll work on updating the homepage such that the data is populated via the database

File: app/components/FoodSection.tsx (hunk 5 above)
Comment (salamf) [reply app/components/FoodSection.tsx:76]: Oops yeah I think I missed that haha, will update.

File: app/components/CategorySection.tsx (hunk 4 above)
Comment (salamf) [reply app/components/CategorySection.tsx:16]: Sounds good! I just had those there temporarily until we got the firebase stuff setup

**Comment by salamf** [comment]
> Overall looks great! Couple high-level questions for ya
> 
> 1. What is the search bar going to do once it gets hooked up? Wondering how it will work from a user and a developer POV. (child components filtering parent data is never an easy implementation). Also wondering if this 'home page' will be for scrolling by menus/restaurants and finding ideas, while another page will offer dedicated macronutrient filters or something??
> 2. What is the down-caret in the layout footer doing? Where does it take you?

1. 
I was thinking it can either be used to search for food items (e.g. spaghetti) and then wed have another page where you could search for specific restaurants (like we had in the original mock up) OR we could make it such that you can search for food and restaurants on that search bar and then just skip the page for searching for restaurants, does that make sense?

And yes, homepage mostly for browsing and stuff, and then another page for a dedicated macronutrient filter search. Also if you have better ideas then we can definitely go with those!

2.
The carrot in the navbar was there before I changed anything, and I'm pretty sure it's supposed to represent the absence of a specified icon in a page. Clicking it now will lead to the test database page you originally created. I'll get rid of it on my next update.

**Comment by johnmck95** [comment]
> I was thinking it can either be used to search for food items (e.g. spaghetti) and then wed have another page where you could search for specific restaurants (like we had in the original mock up) OR we could make it such that you can search for food and restaurants on that search bar and then just skip the page for searching for restaurants, does that make sense?
> 
> And yes, homepage mostly for browsing and stuff, and then another page for a dedicated macronutrient filter search. Also if you have better ideas then we can definitely go with those!
>
> 
> The carrot in the navbar was there before I changed anything, and I'm pretty sure it's supposed to represent the absence of a specified icon in a page. Clicking it now will lead to the test database page you originally created. I'll get rid of it on my next update.

1.  Might be worth thinking about how we want to hook up the search bar carefully for milestone 2 if it's not going to be used to filter by macros (our main use case). I think it might be worth pitching the idea to the mentors that _the home page is mainly for revenue_ - users can filter by foods/restaurants which drive advertisements, and restaurants can pay to be near the top of the home page. The search bar is only for filtering restaurants and food items. Then we have a totally separate macronutrient filtration page which is the core reason people come to the app. If we don't have our main use case on the home page, i think we'll need to be able to explain why not

2. Perfect!

**Review by mehradadimi — APPROVED** [review]

**Pull request merged**